geomstats/geometry/general_linear.py

```diff
@@ -1,192 +1,125 @@
 """Module exposing the GeneralLinear group class."""
 
 import geomstats.backend as gs
-from geomstats.geometry.lie_group import LieGroup
 from geomstats.geometry.matrices import Matrices
 
 
-class GeneralLinear(LieGroup, Matrices):
-    """Class for the General Linear Group, i.e. the matrix group GL(n).
-
-    Note: The default representation for elements of GL(n)
-    are matrices. For now, SO(n) and SE(n) elements are represented
-    by a vector by default.
-    """
+class GeneralLinear(Matrices):
+    """Class for the general linear group GL(n)."""
 
     def __init__(self, n):
```

Comment:
You do not instantiate from the parent class here?

Reply:
why did you delete the assert?

Reply:
Good question, I'll restore `Matrices.__init__(self, n, n)` so that the metric is set to the Frobenius metric 
`assert` will then be redundant as matrices already check n > 0, but this not proper error catching either btw... nothing is printed!

Reply:
I checked the `lie_group.py` file but it does not seem a good idea to call its `__init__` method: this would override the metric to `InvariantMetric(...)` which is not defined